tutor/src/components/annotations/annotation.jsx

```diff
@@ -0,0 +1,353 @@
+
+  return {
+    bottom: rect.bottom + wTop,
+    top: rect.top + wTop,
+    left: rect.left + wLeft,
+    right: rect.right + wLeft
+  }
+}
+
+const HighlightWidget = ({style, annotate, highlight}) => (
+  style ?
+    <div className="widget arrow_box" style={style}>
+      <i className="fa fa-comment" role="button" alt="annotate" onClick={annotate}></i>
+      <i className="fa fa-pencil" role="button" alt="highlight" onClick={highlight}></i>
+    </div>
+  : null
+);
+
+const EditBox = (props) => {
+  return (
+    <div className={`edit-box ${props.show}`}>
+      <textarea value={props.annotation} onChange={props.updateAnnotation}></textarea>
+      <div className="button-row">
+        <div className="button-group">
+          <button aria-label="save" className="primary" onClick={props.save}>
+            <i className="fa fa-check"></i>
+          </button>
+          <button aria-label="delete" className="secondary" onClick={props.delete}>
+            <i className="fa fa-trash"></i>
+          </button>
+        </div>
+        <div className="button-group">
+          <button aria-label="previous annotation" onClick={props.previous}>
+            <i className="fa fa-chevron-up" role="button" ></i>
+          </button>
+          <button aria-label="next annotation">
+            <i className="fa fa-chevron-down" role="button" onClick={props.next}></i>
+          </button>
+          <button>See all</button>
+        </div>
+      </div>
+    </div>
+  );
+};
+
+const SidebarButtons = ({items, offsetTop, onClick}) => (
+  <div>
+    {items.filter((item) => item.rect).map((item, index) => {
+      const style = {
+        right: 0,
+        top: item.rect.top - offsetTop,
+        position: 'absolute'
+      };
+
+      return item.annotation.length ?
+        <button className="sidebar-button" style={style} key={item.selection.start} onClick={() => onClick(item)}>
+          <i className="fa fa-comment"></i>
+        </button>
+        :
+        null;
+    })}
+  </div>
+);
+
+export default class AnnotationWidget extends React.PureComponent {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      activeHighlight: null,
+      widgetStyle: {display: 'none'},
+      preload: [],
+      needToLoadAnnotations: true
+    };
+  }
+
+  componentDidMount() {
+    this.handleSelectionChange = _.debounce(() => {
+      if (!this.state.activeHighlight) {
+        this.setWidgetStyle();
+      }
+    }, 80);
+    window.document.addEventListener('selectionchange', this.handleSelectionChange)
+    this.restoreAnnotations();
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.setState({
+      needToLoadAnnotations: nextProps.documentId !== this.props.documentId
+    });
+  }
+
+  componentDidUpdate() {
+    this.restoreAnnotations();
+  }
+
+  componentWillUnmount() {
+    window.document.removeEventListener('selectionchange', this.handleSelectionChange)
+  }
+
```

Comment:
A good practice is to not reference `window` directly.  If you instead add a `windowImpl` (or whatever) prop that defaults to `window` you can then mock the window in specs to make sure the component responds appropriately.

Example:
https://github.com/openstax/tutor-js/blob/master/tutor/src/components/student-preview/builder-popup.jsx#L18
and
https://github.com/openstax/tutor-js/blob/master/tutor/specs/components/student-preview/builder-popup.spec.jsx#L13

I'm fine with adding that later though, just wanted you to be aware of it 😁